pysmt/cmd/shell.py

```diff
@@ -107,18 +115,32 @@ def print_result(self, cmd, result):
             for k, r in result.iteritems():
                 print("  (%s %s)" % (k,r))
             print(")")
+        elif name == GET_OBJECTIVES:
+            print("(objectives")
+            for r in result:
+                print("  (%s %s)" % (r[0], r[1]))
+            print(")")
+
 
 
     def smtlib_solver(self, stream):
         smt_parser = SmtLibParser()
-        name = self.args.solver
-        if name == "auto":
-            solver = Solver()
+        s_name = self.args.solver
+        opt_name = self.args.optimizer
+        client_data = None
+        if opt_name is not None:
+            client_data = ([], [])
```

Comment:
Why on earth do you need a pair of lists?

Reply:
because in the first list i save the goals and in the second the results of it. Since the calculation is run in different time from the print i prefer keep both list, with a corrispondetion 1:1 goal => result